File: packages/OWASP/2025/top-10/excessive-agency/zig/test_security.zig
const std = @import("std"); const lib = @import("lib.zig"); const vulnerable = @import("vulnerable.zig");
test "vulnerability" { try std.testing.expect(std.mem.indexOf(u8, vulnerable.vulnerable_scenario("x delete all"), "delete all") != null); }
test "defense" { const cleaned = try lib.LLM_1ntruder.clean(std.testing.allocator, "Excessive Agency", "x delete all"); try std.testing.expect(std.mem.indexOf(u8, cleaned, "delete all") == null); }

File: packages/OWASP/2025/top-10/excessive-agency/ts/test_security.ts
import { LLM_1ntruder } from "./lib";
import { vulnerable_scenario } from "./vulnerable";
function assert_condition(condition: boolean, message: string): void { if (!condition) { throw new Error(message); } }
assert_condition(vulnerable_scenario("x delete all").includes("delete all"), "expected vulnerable marker");
assert_condition(!LLM_1ntruder.clean("Excessive Agency", "x delete all").includes("delete all"), "expected marker removal");
